_posts/2021-07-19-dd_00002_transistor.md

```diff
@@ -0,0 +1,236 @@
+
+Let's see what goes on inside a transistor.
+
+When there is no voltage present at the gate, the channel between source and
+drain is not conductive, i.e., no current can flow from source to drain. The
+MOSFET is switched _off_. The following picture depicts this state (notice how
+the electrons are waiting at the source terminal):
+
+<figure class="img-center" style="width:50%">
+  <img src="/assets/images/dd_00002-mosfet-channel-off.png" alt="">
+  <!-- <figcaption class="figure-caption text-center">Microchips</figcaption> -->
+</figure>
+
+When we now start applying a _positive_ voltage at the gate terminal, the gate
+will attract **electrons** from the source terminal (remember that the source is
+made from Silicon which has some free electrons). As we increase the gate
+voltage, more and more electrons from the source will _accumulate_ near the gate
+(the area where the channel is supposed to be). Note that the electrons actually
+would love to get to the positively charged gate, but they can't because of the
+insulating oxide layer between gate and bulk. When the gate voltage reaches the
+_threshold voltage_, there are now enough electrons at the channel area to allow
+a current flow between source and drain. Imagine the accumulated electrons in
+the channel as a "bridge" between source and drain. Now electrons can flow from
+source to drain; the MOSFET is switched _on_:
+
+<figure class="img-center" style="width:50%">
+  <img src="/assets/images/dd_00002-mosfet-channel-on.png" alt="">
+  <!-- <figcaption class="figure-caption text-center">Microchips</figcaption> -->
+</figure>
+
+There are two types of MOSFETs. The one we presented you above is also referred
+to as **NMOS**. The other MOSFET type is called **PMOS**. The difference between
+the two is that the PMOS has the opposite behavior of an NMOS: when there is
+_no_ gate voltage, the PMOS is switched on, and a positive gate voltage switches
+off the PMOS. We will spare you further details here[^2], but for now, it's
+sufficient to know that there are these two types of MOSFETs, and they
+essentially work opposite to each other. In digital systems, we need both NMOS
+and PMOS to build complex circuits.
+
+We hope that by now, you have a broad understanding of how transistors work. If
+you want to see a more visual explanation and maybe also understand the more
+low-level details of how MOSFETs work (e.g., why we have to "dope" Silicon to
+enhance its conductivity), check out this great
+[video](https://www.youtube.com/watch?v=IcrBqCFLHIY) by Veritasium. Another
+great explanatory video with animations is
+[this](https://www.youtube.com/watch?v=Bfvyj88Hs_o) one.
+
+It shall be noted that even though it is helpful (and sometimes necessary) to
+understand the detailed working of transistors, in many practical situations,
+especially when designing digital circuits, it is often enough to just know that
+transistors can be in two states, _on_ or _off_. This is one example of an
```

Comment:
```suggestion
transistors can be in two states, _on_ or _off_, and that there is a PMOS and an NMOS type. This is one example of an
```

Reply:
Not sure about this one.. Firstly, I had this (that it's enough to know there are PMOS/NMOS, and they are basically opposite) already some lines earlier. And secondly, I thinks this would break the bridge to the abstraction sentence.

I would prefer the current way, because it's also general ("on/off" equally applies to both NMOS and PMOS). What do you think?